volatility3/framework/plugins/windows/pe_symbols.py

```diff
@@ -0,0 +1,777 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+
+
+import io
+import logging
+
+from typing import Dict, Tuple, Optional, List, Generator, Union
+
+import pefile
+
+from volatility3.framework import interfaces, exceptions
+from volatility3.framework import renderers, constants
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints
+from volatility3.framework.symbols import intermed
+from volatility3.framework.symbols.windows import pdbutil
+from volatility3.framework.symbols.windows.extensions import pe
+from volatility3.plugins.windows import pslist, modules
+
+vollog = logging.getLogger(__name__)
+
+
+class PESymbolFinder:
+    """
+    Interface for PE symbol finding classes
+    This interface provides a standard way for the calling code to
+    lookup symbols by name or address
+    """
+
+    cached_str = Union[str, None]
+    cached_str_dict = Dict[str, cached_str]
+
+    cached_int = Union[int, None]
+    cached_int_dict = Dict[str, cached_int]
+
+    cached_value = Union[int, str, None]
+    cached_value_dict = Dict[str, Union[Dict[str, List[str]], Dict[str, List[int]]]]
+
+    def __init__(
+        self,
+        layer_name: str,
+        mod_name: str,
+        module_start: int,
+        symbol_module: Union[interfaces.context.ModuleInterface, pefile.ExportDirData],
```

Comment:
They both inherit from PESymbolFinder. This is how to type check interfaces.

Reply:
Well, then the type here should be PESymbolFinder?  Doing the Union is basically saying "The signatures differ but I'm allowing both", otherwise you'd just use the interface...

Reply:
Fixed